tutorials/Tutorial MIS & QAOA.ipynb

```diff
@@ -0,0 +1,394 @@
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import networkx as nx\n",
+    "A  = np.matrix([[0, 1, 1, 1, 1], [1, 0, 1, 0, 0], [1, 1, 0, 0, 0], [1, 0, 0, 0, 0], [1, 0, 0, 0, 0]])\n",
+    "G_nx = nx.from_numpy_matrix(A)\n",
+    "fig,ax1 = plt.subplots(figsize = (4,4))\n",
+    "nx.draw(G_nx,\n",
+    "        pos,\n",
+    "        edge_color='blue', \n",
+    "        ax=ax1,\n",
+    "        width=1,\n",
+    "        linewidths=1,\n",
+    "        node_size=250,\n",
+    "        node_color='g',\n",
+    "        alpha=1.,\n",
+    "        labels={node:node for node in G_nx.nodes()}\n",
+    "        )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This graph has two maximal independent sets: $(1,3,4)$ and $(2,3,4)$, respectively $'10100'$ and $'11000'$ in binary. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 2 Build the quantum loop "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, we must build the quantum part of the QAOA. All atoms are initially in the groundstate $|00\\dots0\\rangle$ of the `ground-rydberg`basis.  We then apply $p$ layers of alternating non-commutative Hamiltonians. The first one, called the mixing Hamiltonian $H_M$, is realized by taking $\\Omega = 1$ MHz, and $\\delta = 0 $ Mhz in the Hamiltonian equation. The second Hamiltonian $H_c$ is realized with $\\Omega = \\delta = 1$ MHz. $H_M$ and $H_c$ are applied turn in turn with parameters $\\tau$ and $t$ respectively. A classical optimizer is then used to estimate the optimal parameters. \n",
+    "\n",
+    "Experimentally, we don't have access to the state vector $\\psi$. We therefore make it more realistic by taking samples from the state vector that results from running the simulation with `simul.run()`. This is done with the built-in method `results.sample_final_state()`, in which we add the measurement basis which was declared at the end of the sequence, and the number of samples desired. Currently, the repetition rate of the machine is $5$Hz."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def quantum_loop(param, r=reg):\n",
+    "    seq = Sequence(r, Chadoq2)\n",
+    "    seq.declare_channel('ch0','rydberg_global')\n",
+    "    middle = int(len(param)/2)\n",
+    "    param = np.array(param)*1 #wrapper \n",
+    "    t = param[:middle] #associated to H_c\n",
+    "    tau = param[middle:] #associated to H_0\n",
+    "    p = len(t)\n",
+    "    for i in range(p):\n",
+    "        pulse_1 = Pulse.ConstantPulse(tau[i], 1., 0, 0) # H_M \n",
+    "        pulse_2 = Pulse.ConstantPulse(t[i], 1., 1., 0) # H_M + H_c \n",
+    "        seq.add(pulse_1, 'ch0')\n",
+    "        seq.add(pulse_2, 'ch0')\n",
+    "    seq.measure('ground-rydberg')\n",
+    "    simul = Simulation(seq, skip=100)\n",
```

Comment:
The `skip` parameter is now replaced by `sampling_rate` and takes values between 0.05 and 1. I recommend you put `sampling_rate=0.05` to get the fastest simulation.

Reply:
taking `sampling_rate =0.05` gives the following error: `tlist not uniform?`

Reply:
Yes, that was actually a Warning from Qutip and it was annoying me about taking uniform random samples. The method just sticks now to  call `np.linspace` and distribute the indices according to `sampling_rate`. I hope you won't get that line anymore if you try `sampling_rate=0.05`.